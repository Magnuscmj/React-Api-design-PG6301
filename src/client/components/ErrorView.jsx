//This file has code from the lecturer and has been changed to fit this assignment
//https://github.com/kristiania-pg6301-2021/pg6301-react-and-express-lectures

import React from "react";
import GoogleButton from "react-google-button";
import {Link} from "react-router-dom";

export function ErrorView({ error }) {
  if (error.status === 401) {
    return (
      <div>
        <h2 className="title">You are not signed in:</h2>
        <Link to="/login">Click here to go to Login</Link>
      </div>
    );
  }
  return <div>An error occurred: {error.toString()}</div>;
}
