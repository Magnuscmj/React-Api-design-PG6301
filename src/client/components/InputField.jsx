//This file has code from the lecturer and has been changed to fit this assignment
//https://github.com/kristiania-pg6301-2021/pg6301-react-and-express-lectures

import React from "react";

export function InputField({ label, onChangeValue, value, type = "text" }) {
  return (
      <div>
        <label>
          {label}{" "}
          <input
              type={type}
              value={value}
              placeholder={type}
              onChange={(e) => onChangeValue(e.target.value)}
          />
        </label>
      </div>
  );
}
