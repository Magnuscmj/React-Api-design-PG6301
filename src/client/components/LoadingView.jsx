//This file has code from the lecturer and has been changed to fit this assignment
//https://github.com/kristiania-pg6301-2021/pg6301-react-and-express-lectures

import React from "react";

export function LoadingView() {
  return <div>Loading...</div>;
}
