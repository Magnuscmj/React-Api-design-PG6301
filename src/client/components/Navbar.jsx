import React from "react";
import { Link } from "react-router-dom";

export function Navbar() {

  return (
    <nav className="navbar navbar-expand-md navbar-dark bg-dark mb-4">
      <div className="container-fluid">
        <div>
          <ul className="navbar-nav me-auto mb-2 mb-md-0">
            <li className="nav-item">
              <Link to="/login" className="nav-link active" aria-current="page">
                Login
              </Link>
            </li>
            <li className="nav-item"></li>
            <li className="nav-item">
              <Link
                to="/profile"
                className="nav-link active"
                aria-current="page"
              >
                Messages
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
