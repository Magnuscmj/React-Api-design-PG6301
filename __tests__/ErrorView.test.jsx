//This file has code from the lecturer and has been changed to fit this assignment
//https://github.com/kristiania-pg6301-2021/pg6301-react-and-express-lectures

import React from "react";
import * as ReactDOM from "react-dom";
import {MemoryRouter} from "react-router";
import {ErrorView} from "../src/client/components/ErrorView";

describe("ErrorView", () => {
    it("should handle error messages",() => {
        const container = document.createElement("div");
        let ok;
        ReactDOM.render(
            <MemoryRouter>
                <ErrorView error={status === ok} />,
            </MemoryRouter>,
            container
        );
        expect(container.innerHTML).toMatchSnapshot();
    })
})