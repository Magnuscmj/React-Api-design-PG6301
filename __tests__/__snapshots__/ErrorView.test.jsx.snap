// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView should handle error messages 1`] = `"<div>An error occurred: false</div>,"`;
